FutureNorthants/northants-website
# Plugged autocomplete search suggestions in

### Setup
This PR relies upon two others so there's a bit of work to do to get it all up and running:

**Backend**
PR: https://github.com/FutureNorthants/northants-website/pull/170
This provides the graphql schema update and associated query functionality that will provide search suggestion data.
Once checked out, `make stop` and `make start` your backend containers.

**Design System**
PR: https://github.com/FutureNorthants/northants-design-system/pull/71
This provides the new Autocomplete component that the front end will use. Once checked out, you'll need to locally publish a copy for your frontend instance to use with `yalc publish`.

**Front End**
Stop your local instance if it's running. Check out this branch. Pull in the local copy of the design system with `yalc add northants-design-system` and then run `npm install` to make sure the new dependencies introduced by the design system PR are pulled in. Now run `yarn dev` to start the front end server. Open a new terminal window and go back into your frontend folder. Run `yarn codegen`, it should run OK (if it doesn't, talk to me right away). If you run `git status` afterwards there shouldn't be any changes in __generated folders (you can ignore the package file changes it'll show) - if there any changes let me know right away again.

You should be now running the new front end changes with the new autocomplete design system component in place.

### Testing
1. Visit the homepage. Should work as normal, but the search box will give you autocomplete suggestions as you type.
2. Click on a suggestion. Should go to the search page with appropriate results. Changing the content of the search box should again show autocomplete suggestions.
3. Go to any other page and expect the search bar at top right to work as normal, but with added autocomplete functionality.
4. Go to the `/news` page and expect the search input under 'search articles' to have autocomplete functionality that gives back suggestions that work when searching news articles (i.e. different to the other suggestions you've seen on the homepage and at the top of other pages).

### Deployment Notes
Deploy backend changes first. We'll then need to publish a design system update and also add another commit to this PR to update the package files correctly (_not_ using the local design system copy!). Finally, deploy this PR.
